Preferences.sublime-settings

```diff
@@ -0,0 +1,3 @@
+{
+        "index_exclude_patterns": ["*.md", "runners", "vendor", "docs", "mock_*.c", "preprocess", "cache", "results", "artifacts", "release", "support", "simulation"]
```

Comment:
This will affect the user's preferences and I can assure you that I'd be very much opposed to having markdown fils or "docs" folders excluded from the index. Please reconsider this setting.

Ideally, users should ignore folders they don't care about on a per-project basis.

Reply:
Can you clarify on the "dumpster diving"? 

In general, if you know you don't want it to process certain folders ever, there is the global `folder_exclude_patterns` setting (similar to the index one here) that users are free to tweak and probably should because it doesn't include `node_packages` by default, for instance, which is a big offender of the Black Hole Syndrome™. However, since these folder names are wildly different across different toolchains, it's not trivial to come up with generally applicable defaults for all situations.

Reply:
https://en.wikipedia.org/wiki/Dumpster_diving
>"[salvaging](https://en.wiktionary.org/wiki/salvage) from large commercial, residential, industrial and construction containers for unused items discarded by their owners but deemed useful to the picker. It is not confined to [dumpsters](https://en.wikipedia.org/wiki/Dumpster) and [skips](https://en.wikipedia.org/wiki/Skip_(container)) specifically and may cover standard household waste containers, curb sides, landfills or small dumps. 

Specifically: "unused items discarded by their owners but deemed useful to the picker".